### PR TITLE
Report line:column instead of index, on failure (#46)

### PR DESCRIPTION
**Pull request by Alain-Bearez**

### CODE REVIEW

File: fastparse/shared/src/main/scala/fastparse/core/Parsing.scala [hunk 1]
```diff
@@ -78,13 +78,16 @@ object Result{
...
+      Nil, input, index, Failure.formatParser(lastParser, input, index)
     )
     override def toString = s"Failure($msg)"
   }
   object Failure {
-    def formatParser(p: Precedence, i: Int) = {
-      Precedence.opWrap(p, Precedence.`:`) + ":" + i
+    def formatParser(p: Precedence, input: String, index: Int) = {
+      val lines = input.take(1 + index).lines.toVector
+      val line = lines.length
```
Comment (lihaoyi) [inline fastparse/shared/src/main/scala/fastparse/core/Parsing.scala]: While you're at this, can you add `line` and `col` onto the `Failure` class as a lazy val?

File: fastparse/shared/src/main/scala/fastparse/core/Parsing.scala [hunk 2]
```diff
@@ -152,8 +159,12 @@ object Result{
      * the default error message isn't to your liking.
      */
     lazy val trace = {
-      Failure.formatStackTrace(stack, input, index, Failure.formatParser(expected0, index))
+      Failure.formatStackTrace(stack, input, index, Failure.formatParser(expected0, input, index))
     }
+
+    private lazy val lines = input.take(1 + index).lines.toVector
```
Comment (lihaoyi) [inline fastparse/shared/src/main/scala/fastparse/core/Parsing.scala]: No need to have it on `TracedFailure`; you already have it on `Failure`, and if you have a `TracedFailure` you have a `Failure`

**Comment by lihaoyi** [comment]
can you rebase this? I'm not sure what workflow you're using that causes all these merge errors, but it sure is making a lot of them =P

**Comment by Alain-Bearez** [comment]
I started four distinct branches in order to tackle four distinct issues. Each time you merge one of them, the changes I had to replicate through all of them, such as excluding the SocketHandler from lila, generate a conflict with the same change based on what is now a different parent branch.

I don't mind doing a last rebase once you reviewed the changes up to your expectations.

**Comment by lihaoyi** [comment]
This looks good to me! Thanks =D

**Pull request merged**